fastexcel-writer/src/test/java/org/dhatim/fastexcel/Correctness.java

```diff
@@ -519,4 +523,69 @@ public void canHideColumns() throws Exception {
         assertFalse("Column 4 should be visible", xws.isColumnHidden(4));
     }
 
+    @Test
+    public void shouldSetAllDefaultSheetProtectionOptions() throws IOException {
+
+        byte[] data = writeWorkbook(wb -> {
+            Worksheet ws = wb.newWorksheet("Worksheet 1");
+            ws.protect("HorriblePassword");
+        });
+
+        // Check generated workbook with Apache POI
+        XSSFWorkbook xwb = new XSSFWorkbook(new ByteArrayInputStream(data));
+        XSSFSheet xws = xwb.getSheetAt(0);
+
+        // Only the defaults are locked.
+        assertThat(xws.isAutoFilterLocked()).isTrue();
+        assertThat(xws.isDeleteColumnsLocked()).isTrue();
+        assertThat(xws.isDeleteRowsLocked()).isTrue();
+        assertThat(xws.isFormatCellsLocked()).isTrue();
+        assertThat(xws.isFormatColumnsLocked()).isTrue();
+        assertThat(xws.isFormatRowsLocked()).isTrue();
+        assertThat(xws.isInsertColumnsLocked()).isTrue();
+        assertThat(xws.isInsertHyperlinksLocked()).isTrue();
+        assertThat(xws.isInsertRowsLocked()).isTrue();
+        assertThat(xws.isPivotTablesLocked()).isTrue();
+        assertThat(xws.isSortLocked()).isTrue();
+        assertThat(xws.isSheetLocked()).isTrue();
+
+        // The rest is not locked
+        assertThat(xws.isObjectsLocked()).isFalse();
+        assertThat(xws.isScenariosLocked()).isFalse();
+        assertThat(xws.isSelectLockedCellsLocked()).isFalse();
+        assertThat(xws.isSelectUnlockedCellsLocked()).isFalse();
+    }
+
+    @Test
+    public void shouldSetSpecificSheetProtectionOptions() throws IOException {
+
+        byte[] data = writeWorkbook(wb -> {
+            Worksheet ws = wb.newWorksheet("Worksheet 1");
+            ws.protect("HorriblePassword", SHEET, DELETE_ROWS, OBJECTS);
```

Comment:
It would be nice to [check the password hash](https://poi.apache.org/apidocs/dev/org/apache/poi/xssf/usermodel/XSSFSheet.html#validateSheetPassword-java.lang.String-) with Apache POI.